### ISSUE TITLE
Save video not working

### ISSUE THREAD

**Issue by Netantican** [xbmc-adult/xbmc-adult#440]
Save video not working on p%rnhub with Kodi 19.3 android TV + videodevil .20. Not sure if it doesn’t work for the other sites, but it did work in .19 before p$rnhub broke, and has been intermittently working in versions previous. Is their site doing something to stop it saving of videos from working or is it a problem on video devil’s end?

When you try to save, it doesn’t produce an error, it just saves a small 200 byte file in the save directory and goes back to the video listing. Can anyone replicate this issue? Can a fix be issued?

thank you

**Comment by Lamoboy** [comment]
I have the same issue with P%rnHub on iOS. From other websites, VideoDevil downloads videos normally.

**Comment by Netantican** [comment]
Shitty :( guess it’s a problem on their end then. Wonder if this plug in can be reconfigured to work with it

**Comment by Gujal00** [comment]
Save Video will only work for mp4, mkv, and flv files. If the site switched to HLS (.m3u8) then save video wont work. it will just get the small m3u8 file in that case

**Comment by Netantican** [comment]
> Save Video will only work for mp4, mkv, and flv files. If the site switched to HLS (.m3u8) then save video wont work. it will just get the small m3u8 file in that case

How do we find out if this is the issue? I feel like it’s not because this issue has happened before and resolved itself but how would we know for sure and why can’t HLS be downloaded? Thanks n

**Comment by Gujal00** [comment]
HLS is not a single file like mkv, mp4, flv, or avi. It is a manifest (.m3u8) and multiple segments (.ts) designed for streaming, You cannot save HLS from a web browser for the same reason.

**Comment by Lamoboy** [comment]
> 
> 
> HLS is not a single file like mkv, mp4, flv, or avi. It is a manifest (.m3u8) and multiple segments (.ts) designed for streaming, You cannot save HLS from a web browser for the same reason.

Actually, you can. There is a plugin for Firefox called "Video DownloadHelper" and it saves HLS streams.

**Comment by Gujal00** [comment]
Exactly, so if someone writes an addon for kodi to download hls the same way someone has written that for a browser, it will work. It is not a standard function in a browser, and is not a standard function in Kodi Python.
Pornhub has switched to HLS streams about a month ago and that is why playback was broken earlier and fixed in the last release to look for hls streams

**Comment by Netantican** [comment]
Actually I am finding this issue with other sites besides pronhbu too. I’m not sure if there’s anyone maintaining this plug in any more, but it would be appreciated if it could be at least looked into to see if this HLS thing is the problem for sure.
